lib/administrate/generator_helpers.rb

```diff
@@ -4,10 +4,29 @@ def call_generator(generator, *args)
       Rails::Generators.invoke(generator, args, generator_options)
     end
 
+    def find_routes_file
+      routes_file = Rails.root.join("config/routes.rb")
+      if !File.file?(routes_file)
+        routes_file = find_file("routes.rb")
+        raise "Unable to locate your routes.rb file" unless !routes_file.nil?
+      end
+      routes_file
+    end
+
     private
 
     def generator_options
       { behavior: behavior }
     end
+
+    def find_file(file_name)
+      file = nil
+      Find.find(Rails.root) do |path|
+        Find.prune if path.include? ".git"
```

Comment:
Do you think there could ever be other similar directories we should be ignoring here? I wonder if we could be more selective with what we're looking for to help avoid this check. What do you think?

Reply:
Hmm now that I think about it, we should probably avoid vendor, test, spec, and public, maybe db too?

I can't really think of another way to avoid the check unless maybe we only look through the entire config and app tree and then "give up" if routes.rb is not there. What are your thoughts on that?

Reply:
Yeah, I'd agree on both of those. I'd probably create an array of possible excludes, add those to a constant and then check that for presence.